src/main/scala/org/ergoplatform/http/api/ErgoUtilsApiRoute.scala

```diff
@@ -33,17 +42,29 @@ class ErgoUtilsApiRoute(ergoSettings: ErgoSettings)(implicit context: ActorRefFa
       )
   }
 
-  def addressToRaw: Route = (get & path("addressToRaw" / Segment)) { addressStr =>
+  def addressToRawR: Route = (get & path("addressToRaw" / Segment)) { addressStr =>
     ergoAddressEncoder.fromString(addressStr)
-        .map(address => address.contentBytes)
-        .map(Base16.encode)
-        .fold(
-          e => BadRequest(e.getMessage),
-          raw => ApiResponse(Map("raw" -> raw).asJson)
-        )
+      .map(address => address.contentBytes)
+      .map(Base16.encode)
+      .fold(
+        e => BadRequest(e.getMessage),
+        raw => ApiResponse(Map("raw" -> raw).asJson)
+      )
+  }
+
+  def ergoTreeToAddressR: Route = (get & path("ergoTreeToAddress" / Segment)) { ergoTreeHex =>
+    Base16.decode(ergoTreeHex)
+      .flatMap { etBytes =>
+        ergoAddressEncoder.fromProposition(
+          treeSerializer.deserializeErgoTree(etBytes).toProposition(replaceConstants = false))
```

Comment:
.toProposition(replaceConstants = false) is not needed, and I'm not sure why you are doing this step.